edgelet/iotedged/src/lib.rs

```diff
@@ -168,7 +168,10 @@ impl Main {
         Main { settings }
     }
 
-    pub fn run(self) -> Result<(), Error> {
+    pub fn run_until<F>(self, shutdown_signal_generator: F) -> Result<(), Error>
+    where
+        F: Fn() -> Box<dyn Future<Item = (), Error = ()> + Send>,
```

Comment:
Is it possible to do something like:

```rust
pub fn run_until<F, G>(self, shutdown_signal_generator: G) -> Result<(), Error>
where
    F: Future<Item = (), Error = ()> + Send>,
    G: Fn() -> F,
{
}
```

This would remove forcing a `box`.

Reply:
Super cool!